Description: Fix reproducibility issues with autogsdoc.

Based on https://sources.debian.org/src/gnustep-base/1.31.1-4/debian/patches/autogsdoc-reproducibility.patch

Upstream status: https://github.com/gnustep/libs-base/issues/697
When updating repalcce GNUSTEP_TZ with SOURCE_AUTHOR_CREDIT.

---

Index: gnustep-base-1.31.1/Tools/AGSOutput.m
===================================================================
--- gnustep-base-1.31.1.orig/Tools/AGSOutput.m
+++ gnustep-base-1.31.1/Tools/AGSOutput.m
@@ -76,6 +76,19 @@ static BOOL snuggleStart(NSString *t)
   return [set characterIsMember: [t characterAtIndex: [t length] - 1]];
 }
 
+static NSString *
+guixuser (void)
+{
+  NSString *user;
+
+  if (getenv("SOURCE_DATE_EPOCH"))
+    user = @"Guix";
+  else
+    user = NSFullUserName();
+
+  return user;
+}
+
 /**
  * <unit>
  *  <heading>The AGSOutput class</heading>
@@ -426,7 +439,7 @@ static BOOL snuggleStart(NSString *t)
   authors = [info objectForKey: @"authors"];
   if (authors == nil)
     {
-      tmp = [NSString stringWithFormat: @"Generated by %@", NSFullUserName()];
+      tmp = [NSString stringWithFormat: @"Generated by %@", guixuser()];
       [str appendString: @"    <author name=\""];
       [str appendString: tmp];
       [str appendString: @"\"></author>\n"];
@@ -2407,7 +2420,7 @@ static BOOL snuggleStart(NSString *t)
 	  [str appendString: kind];
 	  [str appendString: @"</title>\n"];
 	  tmp = [NSString stringWithFormat: @"Generated by %@",
-	    NSFullUserName()];
+	    guixuser()];
 	  [str appendString: @"    <author name=\""];
 	  [str appendString: tmp];
 	  [str appendString: @"\"></author>\n"];
Index: gnustep-base-1.31.1/Tools/AGSParser.m
===================================================================
--- gnustep-base-1.31.1.orig/Tools/AGSParser.m
+++ gnustep-base-1.31.1/Tools/AGSParser.m
@@ -38,6 +38,25 @@
 
 #define	ENDBRACE	0x7D	// '}' character
 
+@interface NSDate (Guix)
++ (instancetype) guixDate;
+@end
+
+@implementation NSDate (Guix)
+
++ (instancetype) guixDate
+{
+
+  NSString *timestamp = [[[NSProcessInfo processInfo] environment]
+			  		       objectForKey: @"SOURCE_DATE_EPOCH"];
+  if (!timestamp)
+    return [NSDate date];
+
+  return [NSDate dateWithTimeIntervalSince1970: [timestamp floatValue]];
+}
+
+@end
+
 /**
  *  The AGSParser class parses Objective-C header and source files
  *  to produce a property-list which can be handled by [AGSOutput].
@@ -2406,7 +2425,7 @@ fail:
       if (nil == generated)
 	{
 	  generated = [[NSString alloc] initWithFormat:
-	    @"<date>Generated at %@</date>", [NSDate date]];
+	    @"<date>Generated at %@</date>", [NSDate guixDate]];
 	}
       [info setObject: generated forKey: @"date"];
     }
